### DISCUSSION TITLE
Deprecated API warning in GKE

### DISCUSSION THREAD

**Discussion by dlouvier** [strimzi/strimzi-kafka-operator#7253] [Q&A] [answered]
Hello folks,

We're run Strimzi operator (0.30.0), installed via Helm Charts. The operator is installed in a GKE Cluster which was recently upgraded to v1.24 (previously v1.23).

Since the last revision, [we're getting warnings](https://cloud.google.com/kubernetes-engine/docs/deprecations/viewing-deprecation-insights-and-recommendations) as some workload in the cluster is calling deprecated APIs in Kubernetes v1.22 which it doesn't exist anymore in the newer releases.

In concrete these are the calls:

|API Endpoint | User Agent|
|-- | -- | 
|/apis/rbac.authorization.k8s.io/v1beta1/rolebindings | okhttp/3.12.12 |
|/apis/rbac.authorization.k8s.io/v1beta1/roles | okhttp/3.12.12|

We identified via GKE Audit logs that seems to be coming from the cluster-operator as It tries to get the roles bindings for "cluster-entity-topic-operator-role"

```
       "permission": "io.k8s.authorization.rbac.v1beta1.rolebindings.get",
        "resource": "rbac.authorization.k8s.io/v1beta1/namespaces/strimzi/rolebindings/cluster-entity-topic-operator-role"
```

Apart of the warning, we didn't notice any functionality is broken, but we'll like to root cause the issue to eliminate these warnings.

Does anyone idea from where could be coming this?

Best regards,

**Comment by scholzj** [comment]
It does not use these resources. You can just ignore the warnings. You can also check the other related discussions. For example #7157.

**Comment by dlouvier** [comment]
@scholzj Thanks for the super quick answer and sorry I didn't hit the correct key words in my search.

Re-posting the instructions from the other answer: https://github.com/orgs/strimzi/discussions/7157#discussioncomment-3331754

> @scholzj
> It is not about the CRDs but about the RBAC API.
> You can ignore this, it will se only the v1 RBAC APi once you upgrade to a Kubernetes version which supports only the v1 API.
> If you want to disable this, you should be able to do so using the KUBERNETES_BACKWARDSCOMPATIBILITYINTERCEPTOR_DISABLE option of the Fabric8 client: https://github.com/fabric8io/kubernetes-client#configuring-the-client
> But you can in general just ignore it

**Comment by dlouvier** [comment, marked as answer]
If anyone else is coming here, you can pass this to the values.yml file in the helm-chart of strimzi-kafka-operator:

```
---
(...)
extraEnvs:
  - name: KUBERNETES_BACKWARDSCOMPATIBILITYINTERCEPTOR_DISABLE
    value: "true"
(...)
```

That will add the environment variable into the operator pod and it will stop calling to deprecated APIs.

**Comment by ee07dazn** [comment]
@scholzj : I have had the same issue and i am trying to add the variable and set its value using terraform. 

```
  set {
    name = "extraEnvs[0].name"
    value = "KUBERNETES_BACKWARDSCOMPATIBILITYINTERCEPTOR_DISABLE"
  }
  set {
    name = "extraEnvs[0].value"
    value = "true"
}
```
Everytime, I try and set this and deploy the helm charts via terraform, it throws an error :

`

│ Warning: Helm release "strimzi-kafkaconnect" was created but has a failed status. Use the `helm` command to investigate the error, correct it, then run Terraform again.
│
│   with helm_release.strimzi,
│   on strimzi.tf line 1, in resource "helm_release" "strimzi":
│    1: resource "helm_release" "strimzi" {
│
╵
╷
│ Error: rolebindings.rbac.authorization.k8s.io "strimzi-cluster-operator-entity-operator-delegation" already exists
│
│   with helm_release.strimzi,
│   on strimzi.tf line 1, in resource "helm_release" "strimzi":
│    1: resource "helm_release" "strimzi" {
│
╵
ERRO[0080] 1 error occurred:
	* exit status 1
`

and randomly, this `"strimzi-cluster-operator-entity-operator-delegation"` keeps changing to some other value like the cluster-operator.

This probably could be a terraform issue but just pasting in case you have a handle on what's going on ?|

**Reply by scholzj** [reply]
I do not use Helm or Helm through Terraform. So I'm afraid I cannot help you with this, sorry.

**Reply by ee07dazn** [reply]
no worries, thanks anyways!

**Comment by scholzj** [comment]
For anyone coming here only now => Strimzi 0.31.0 and newer uses Fabric8 Kubernetes Client 6.0.0+ where the backwards compatibility interceptor is disabled by default. So you should not need to disable it anymore.
